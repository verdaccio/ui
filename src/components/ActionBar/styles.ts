import styled from 'react-emotion';
import { default as MuiFab } from '@material-ui/core/Fab';

import colors from '../../utils/styles/colors';
import ListItem from '../../muiComponents/ListItem';

export const ActionListItem = styled(ListItem)({
  paddingTop: 0,
  paddingLeft: 0,
  paddingRight: 0,
});

export const Fab = styled(MuiFab)({
  '&&': {
    backgroundColor: colors.primary,
    color: colors.white,
    marginRight: '10px',
  },
});
