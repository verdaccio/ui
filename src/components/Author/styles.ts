import styled from '@emotion/styled';

import { Theme } from 'verdaccio-ui/design-tokens/theme';

import ListItem from '../ListItem';
import ListItemText from '../ListItemText';
import Text from '../Text';

export const StyledText = styled(Text)<{ theme?: Theme }>(props => ({
  fontWeight: props.theme && props.theme.fontWeight.bold,
}));

export const AuthorListItem = styled(ListItem)({
  padding: 0,
  ':hover': {
    backgroundColor: 'transparent',
  },
});

export const AuthorListItemText = styled(ListItemText)({
  padding: '0 10px',
  margin: 0,
});
