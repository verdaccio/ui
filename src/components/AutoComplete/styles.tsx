import styled from '@emotion/styled';
import React from 'react';

import { Theme } from 'verdaccio-ui/design-tokens/theme';

import Paper from '../Paper';
import TextField from '../TextField';

export interface InputFieldProps {
  color: string;
}

export const Wrapper = styled('div')({
  width: '100%',
  height: '32px',
  position: 'relative',
  zIndex: 1,
});

export const StyledTextField = styled(TextField)<{ theme?: Theme }>(props => ({
  '& .MuiInputBase-root': {
    ':before': {
      content: "''",
      border: 'none',
    },
    ':after': {
      borderColor: props.theme && props.theme.palette.white,
    },
    ':hover:before': {
      content: 'none',
    },
  },
  '& .MuiInputBase-input': {
    color: props.theme && props.theme.palette.white,
  },
}));

/* eslint-disable verdaccio/jsx-spread */
// @ts-ignore types of color are incompatible
export const InputField: React.FC<InputFieldProps> = ({ ...others }) => <StyledTextField {...others} />;

export const SuggestionContainer = styled(Paper)({
  maxHeight: '500px',
  overflowY: 'auto',
});
