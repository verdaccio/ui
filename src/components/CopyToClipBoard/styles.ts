import styled from '@emotion/styled';

import IconButton from '../IconButton';

export const ClipBoardCopy = styled('div')({
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'space-between',
});

export const ClipBoardCopyText = styled('span')({
  display: 'inline-block',
  textOverflow: 'ellipsis',
  overflow: 'hidden',
  whiteSpace: 'nowrap',
  height: '21px',
  fontSize: '1rem',
});

export const CopyIcon = styled(IconButton)({});
