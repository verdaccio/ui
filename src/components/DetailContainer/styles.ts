import styled from 'react-emotion';

export const Content = styled('div')`
  && {
    padding: 15px;
  }
`;
