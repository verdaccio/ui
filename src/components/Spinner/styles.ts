import styled from '@emotion/styled';
import { css } from '@emotion/core';

import colors from '../../utils/styles/colors';
import CircularProgress from '../../muiComponents/CircularProgress';

interface WrapperProps {
  centered: boolean;
}

export const Wrapper = styled('div')<WrapperProps>`
  display: flex;
  align-items: center;
  justify-content: center;
  ${({ centered }) =>
    centered &&
    css`
      position: absolute;
      top: 50%;
      left: 50%;
      transform: translate(-50%, -50%);
    `};
`;

export const Circular = styled(CircularProgress)({
  color: colors.primary,
});
