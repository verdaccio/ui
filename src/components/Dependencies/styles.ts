import styled from 'react-emotion';

import { fontWeight } from '../../utils/styles/sizes';
import Text from '../../muiComponents/Text';
import Card from '../../muiComponents/Card';
import Chip from '../../muiComponents/Chip';

export const CardWrap = styled(Card)({
  margin: '0 0 16px',
});

export const StyledText = styled(Text)({
  fontWeight: fontWeight.bold,
  textTransform: 'capitalize',
});

export const Tags = styled('div')({
  display: 'flex',
  justifyContent: 'start',
  flexWrap: 'wrap',
  margin: '0 -5px',
});

export const Tag = styled(Chip)({
  margin: '5px',
});
