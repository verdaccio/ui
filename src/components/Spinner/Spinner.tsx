import React from 'react';

import { Circular, Wrapper } from './styles';

interface Props {
  size?: number;
  centered?: boolean;
}

const Spinner: React.FC<Props> = ({ size = 50, centered = false }) => (
  <Wrapper centered={centered}>
    <Circular size={size} />
  </Wrapper>
);

export default Spinner;
