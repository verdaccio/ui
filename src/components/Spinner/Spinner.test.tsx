import React from 'react';
import { shallow } from 'enzyme';

import Spinner from './Spinner';
import { Circular, Wrapper } from './styles';

describe('<Spinner /> component', () => {
  test('should render the component in default state', () => {
    const wrapper = shallow(<Spinner />);
    const wrapperComponent = wrapper.find(Wrapper);
    const circular = wrapper.find(Circular);

    expect(circular.props().size).toBe(50);
    expect(wrapperComponent.props().centered).toBe(false);
  });

  test('should render the component in custom state', () => {
    const wrapper = shallow(<Spinner centered={true} size={10} />);
    const wrapperComponent = wrapper.find(Wrapper);
    const circular = wrapper.find(Circular);

    expect(circular.props().size).toBe(10);
    expect(wrapperComponent.props().centered).toBe(true);

    expect(wrapper.html()).toMatchSnapshot();
  });
});
