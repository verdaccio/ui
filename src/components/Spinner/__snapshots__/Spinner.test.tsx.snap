// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spinner /> component should render the component in custom state 1`] = `"<div class=\\"css-1yjbhc2\\"><div class=\\"MuiCircularProgress-root css-15gl0ho MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate\\" style=\\"width:10px;height:10px\\" role=\\"progressbar\\"><svg class=\\"MuiCircularProgress-svg\\" viewBox=\\"22 22 44 44\\"><circle class=\\"MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate\\" cx=\\"44\\" cy=\\"44\\" r=\\"20.2\\" fill=\\"none\\" stroke-width=\\"3.6\\"></circle></svg></div></div>"`;
